lib/beaker/shared/fog_file_parser.rb

```diff
@@ -0,0 +1,48 @@
+require 'stringify-hash'
+
+module Beaker
+  module Shared
+    # A set of functions to read .fog files
+    module FogFileParser
+      # Read a .fog file
+      #
+      # @note Loaded .fog files may use symbols for keys.
+      #    Although not clearly documented, this *is* valid:
+      #    https://www.rubydoc.info/gems/fog-core/1.42.0/Fog#credential-class_method
+      #    https://github.com/fog/fog-core/blob/7865ef77ea990fd0d085e49c28e15957b7ce0d2b/spec/utils_spec.rb#L11
+      #
+      # @param fog_file_path [String] dot fog path. Overridden by
+      # @return [StringifyHash] credentials stored in fog_file_path
+      # @raise [ArgumentError] when the specified credential file is invalid, describing the reson
+      def parse_fog_file(fog_file_path = '~/.fog', credential = :default)
+        # respect credential location from env
+        if ENV["FOG_RC"]
+          fog_file_path = ENV["FOG_RC"]
+          from_env = ' set in ENV["FOG_RC"]'
+        end
+        if !File.exist?(fog_file_path)
```

Comment:
I don't know if all this error handling is necessary. Shouldn't YAML.load output an appropriate error message if the fog file does not exist? And if the syntax is incorrect (so that the Psych::SyntaxError would not be necessary)?

Reply:
Wrapping the full variety of typical failure here under ArgumentError is intended to reduce the duplication of error-handling code throughout all calling methods. Previously, consumers performed a variety of inconsistent error handling on their own ([example](https://github.com/puppetlabs/beaker-vmpooler/pull/11/files#diff-ca1d581f345cffd7fd28e3ffa1d50c0bL43)). But to them, it doesn't matter *why* parsing fails, as any way the credentials are not available. Moving error handling to here simplifies the spec of callers so that their tests do not have duplicate coverage of error conditions, which adding appropriate exception handling to all of them would necessitate. I tried to find a balance between modifying the behavior and spec tests of the callers and providing a uniform API for them to all consume. I guess I should prune the consumer error handling and spec tests down to reflect this.

Reply:
Oh ok. Why not something like?
```
begin
  YAML.load_file(fog_file_path)
rescue Psych::SyntaxError, Errno::ENOENT => e
  raise ArgumentError.new("Failed getting credentials from fog file. Reason: #{e.message}")
end
```

Reply:
Sure, that's a nice refactor that simplifies the spec tests.